### DISCUSSION TITLE
Focus the buffer file in the tree while toggling neotree

### DISCUSSION THREAD

**Discussion by semi710** [nvim-neo-tree/neo-tree.nvim#1586] [Q&A] [answered]
for e.g if I open a buffer via `nvim filename.text` and then toggle neotree i want neotree to expand the dir in which the buffer is and the buffer should be selected.

**Comment by pynappo** [comment, marked as answer]
does `:Neotree reveal` do what you want?

Or are you talking about opening a buffer for a file that doesn't exist yet and then displaying it in the filesystem tree as if it existed?

**Reply by semi710** [reply]
`reveal` is the thing that i wanted tysm for the response.

btw is there any plans or way i can get oil like behaviour to edit the file names and stuff?

**Reply by pynappo** [reply]
being able to edit the neo-tree buffer directly is probably not going to happen, at least not without a complete rewrite. would just be way too much code to maintain. i'd like to make it easier to make neo-tree actions closer to vim-like (maybe keymap presets and keymaps that override textobjects and stuff), but still going through the large backlog of bugs as a new maintainer.

in the meantime you're free to rebind the keys in a way that makes more intuitive sense to you or make a bind that opens oil in the directory where the cursor is.

**Reply by semi710** [reply]
reasonable, will use that map technique for oil.
